### PR TITLE
Release/1.4.0 beta.2 pre release actions

### PR DESCRIPTION
**Pull request by ddimatos**
##### SUMMARY
Issue is to track the added changes as part of the release work, this issue not only addressed a number of lint issues it also cleaned up the changelog. 

Things to note, many unused imports were removed and the deprecation notices were changed to allow for non-major release deprecation meaning i changed the deprecation to be at the option level and also removed some of the sftp_port vars since they were a NOOP anyhow and just raising lint issues. Whether a user stets them today our not, the port is a NOOP from our code, ports need to be done through `ansible.cfg`.

Addresses also #523 

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

**Comment by ddimatos** [comment]
image
Log suffix: `/job/zosAnsible/1277/consoleText`

File: CHANGELOG.rst [hunk 1]
```diff
@@ -58,6 +58,35 @@ New Modules
...
+This changelog describes all changes made to the modules and plugins included
+in this collection. The release date is the date the changelog is created.
+For additional details such as required dependencies and availability review
+the collections `release notes <https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/release_notes.html>`__ 
+
+
+Minor Changes
+-------------
+
+- zos_copy - was enhanced for when `src` is a directory and ends with "/", the contents of it will be copied into the root of `dest`. It it doesn't end with "/", the directory itself will be copied. (https://github.com/ansible-collections/ibm_zos_core/pull/515)
```
Comment (fernandofloresg) [inline CHANGELOG.rst:77]: This is a typo "It it doesn't end" should be "If it doesn't end"

File: changelogs/changelog.yaml [hunk 2]
```diff
@@ -251,6 +251,47 @@ releases:
...
+        using option `encoding`. (https://github.com/ansible-collections/ibm_zos_core/pull/462)
+      - zos_ssh - fixes connection plugin which will error when using `ansible-core`
+        2.11 with an `AttributeError module 'ansible.constants' has no attribute 'ANSIBLE_SSH_CONTROL_PATH_DIR'`.
+        (https://github.com/ansible-collections/ibm_zos_core/pull/462)
+      - zos_ssh - fixes connection plugin which will error when using `ansible-core`
+        2.11 with an `AttributeError module 'ansible.constants' has no attribute 'ANSIBLE_SSH_CONTROL_PATH_DIR'`.
+        (https://github.com/ansible-collections/ibm_zos_core/pull/513)
+      minor_changes:
+      - zos_copy - was enhanced for when `src` is a directory and ends with "/", the
+        contents of it will be copied into the root of `dest`. It it doesn't end with
```
Comment (fernandofloresg) [inline changelogs/changelog.yaml:281]: same typo "It it doesn't" should be "If it doesn't"

File: plugins/modules/zos_encode.py [hunk 3]
```diff
@@ -19,6 +19,7 @@
```
Comment (fernandofloresg) [inline plugins/modules/zos_encode.py:19]: Should 2022 also be added in zos_encode copyright header?

File: plugins/modules/zos_ping.rexx [hunk 4]
```diff
@@ -19,7 +19,7 @@
 DOCUMENTATION = '''
```
Comment (fernandofloresg) [inline plugins/modules/zos_ping.rexx:19]: Should 2022 also be added in zos_ping.rexx copyright header?

**Review by fernandofloresg — APPROVED** [review]
Looks good to me, added comments and questions about this, if copyright updates are not needed then merge is approved.

**Comment by ddimatos** [comment]
@fernandofloresg - made all the changes thank you for the comments and yes if the source is touched in any way we need to update the copyright year.

**Pull request merged**